lib/enumerable.rb

```diff
@@ -0,0 +1,111 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+
+module Enumerable
+  # 1.my_each
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+    self
+  end
+
+  # 2.my_each_with_index
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield(to_a[i], i)
+      i += 1
+    end
+    self
+  end
+
+  # 3.my_select
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    to_a.my_each { |item| new_arr << item if yield item }
+    new_arr
+  end
+
+  # 4.my_all?
+  def my_all?(param = nil)
+    if param.class == Class
+      to_a.my_each { |item| return false unless [item.class, item.class.superclass].include?(param) }
+    elsif param.class == Regexp
+      to_a.my_each { |item| return false unless item.match(param) }
+    elsif !block_given?
+      to_a.my_each { |item| return false if item.nil? || !item }
+    else
+      to_a.my_each { |item| return false unless yield item }
+    end
+    true
+  end
```

Comment:
## False when doesn't match the pattern
- Should return false if any of the collection doesn't match the pattern
```ruby
p [3, 3, 2].my_all?(3) 
# => expected: false
# => got : true
```

Reply:
It's now working, I have changed the method.